game-app/game-core/src/main/java/games/strategy/triplea/ui/TechPanel.java

```diff
@@ -292,20 +290,18 @@ private static String getTechListToolTipText(final TechnologyFrontier techCatego
     }
     final Collection<TechAdvance> listedAlready = new HashSet<>();
     final StringBuilder strTechCategory = new StringBuilder("Available Techs:  ");
-    final Iterator<TechAdvance> iterTechList = techList.iterator();
-    while (iterTechList.hasNext()) {
-      final TechAdvance advance = iterTechList.next();
+    for (final TechAdvance advance : techList) {
       if (listedAlready.contains(advance)) {
         continue;
       }
       listedAlready.add(advance);
       final int freq = Collections.frequency(techList, advance);
+      if (strTechCategory.length() > 0) {
+        strTechCategory.append(", ");
+      }
       strTechCategory
           .append(advance.getName())
           .append(freq > 1 ? " (" + freq + "/" + techList.size() + ")" : "");
-      if (iterTechList.hasNext()) {
-        strTechCategory.append(", ");
-      }
     }
     return strTechCategory.toString();
```

Comment:
If I'm not missing something you could use neat stream notation here, something like:
```java
return techList.stream()
  .distinct()
  .map(advance -> {
    final int freq = Collections.frequency(techList, advance);
    return advance.getName() + (freq > 1 ? " (" + freq + "/" + techList.size() + ")" : "");
  })
  .collect(Collectors.joining(", "));
```

Reply:
Done, thanks!